kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java

```diff
@@ -438,9 +495,21 @@ private String getDeviceIpInterface(Device device) throws DBusException {
         return deviceProperties.Get(NM_DEVICE_BUS_NAME, NM_DEVICE_PROPERTY_INTERFACE);
     }
 
-    private Device getDeviceByIpIface(String iface) throws DBusException {
-        DBusPath ifaceDevicePath = this.nm.GetDeviceByIpIface(iface);
-        return this.dbusConnection.getRemoteObject(NM_BUS_NAME, ifaceDevicePath.getPath(), Device.class);
+    private Optional<Device> getDeviceByInterfaceName(String interfaceName) throws DBusException {
+        Optional<Device> device = Optional.empty();
+        if (!interfaceName.startsWith("ppp")) {
```

Comment:
I don't like this.

We already have the `CONFIGURATION_SUPPORTED_DEVICE_TYPES` AND `STATUS_SUPPORTED_DEVICE_TYPES`. Can't we rely on those?

We should remove the `PPP` device type from those.

See:

https://github.com/eclipse/kura/blob/04d3f581e9073e646324208abc30692fec5420ba/kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java#L73-L74

and:

https://github.com/eclipse/kura/blob/04d3f581e9073e646324208abc30692fec5420ba/kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java#L78-L80

Reply:
I totally agree on this. I don't like this implementation.